### ISSUE TITLE
[Question] Can iOS keep bluetooth connection in background?

### ISSUE THREAD

**Issue by riseworld27** [innoveit/react-native-ble-manager#654]
I want to know if iOS can keep bluetooth connection in background.

**Comment by SAAHIIIL** [comment]
Yes. It is possible to keep the BLE Connection alive in the background as well. You check out the code and try it out.

**Comment by riseworld27** [comment]
Is it supported in this library by default?  Could you indicate some part of code to demonstrate background working of BLE?

**Comment by willluk** [comment]
Interested also.  We have added a listener for BleManagerModule NativeEventEmitter.  However, seems like we are not getting any read event when the app goes into background.

**Comment by danielbenedykt** [comment]
+1

**Comment by SAAHIIIL** [comment]
@riseworld27  The connection would be alive when alive when in background, but your reads and writes would fail in that case, you would have to do reads and writes when the app is open.

But ble manager does have such methods but you can check this out.

https://developer.apple.com/library/archive/documentation/NetworkingInternetWeb/Conceptual/CoreBluetooth_concepts/CoreBluetoothBackgroundProcessingForIOSApps/PerformingTasksWhileYourAppIsInTheBackground.html#//apple_ref/doc/uid/TP40013257-CH7-SW10
